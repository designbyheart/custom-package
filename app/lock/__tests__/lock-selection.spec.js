import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import LockSelection from '../lock-selection'

function props() {
  return {
    switchErrorAlerts: jest.fn(),
  }
}

const store = {
  getState() {
    return jest.fn()
  },
  subscribe() {
    return jest.fn()
  },
  dispatch() {
    return jest.fn()
  },
}

describe('app lock selection page should', () => {
  it('render properly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <LockSelection {...props()} />
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
