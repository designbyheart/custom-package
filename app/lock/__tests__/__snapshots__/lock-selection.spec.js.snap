// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app lock selection page should render properly 1`] = `
<View
  senary={true}
  style={
    Object {
      "backgroundColor": "#2D2D2D",
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 50,
    }
  }
>
  <View
    style={
      Object {
        "paddingVertical": 8,
      }
    }
    verticalSpace={true}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={undefined}
      style={
        Array [
          Object {
            "fontSize": 20,
          },
          Object {
            "color": "#ffffff",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
          null,
        ]
      }
      testID={undefined}
    >
      Choose how to unlock this app
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 8,
      }
    }
    verticalSpace={true}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={undefined}
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "color": "#ffffff",
          },
          Object {
            "fontWeight": "500",
          },
          null,
          Object {
            "textAlign": "center",
          },
          null,
        ]
      }
      testID={undefined}
    >
      This application must be protected by TouchId or a pin code at all times.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      center={true}
      hitSlop={undefined}
      onLayout={undefined}
      onPress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ebebea",
          "borderRadius": 13,
          "justifyContent": "center",
          "margin": 10,
          "minHeight": 130,
          "paddingHorizontal": 20,
          "paddingTop": 10,
        }
      }
      tertiary={true}
      testID={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "marginVertical": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "backgroundColor": "transparent",
              "color": "#484848",
            },
            Object {
              "fontWeight": "500",
            },
            null,
            Object {
              "textAlign": "center",
            },
            null,
          ]
        }
        testID={undefined}
      >
        Use Touch ID for "connect.me"
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={undefined}
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            null,
          ]
        }
        testID={undefined}
      >
        or
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onPress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ebebea",
          "borderRadius": 13,
          "margin": 10,
          "minHeight": 130,
          "paddingHorizontal": 20,
          "paddingTop": 10,
        }
      }
      tertiary={true}
      testID={undefined}
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#484848",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={undefined}
            style={
              Array [
                Object {
                  "fontSize": 27,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#484848",
                },
                Object {
                  "fontWeight": "500",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                null,
              ]
            }
            testID={undefined}
          >
            *
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#484848",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={undefined}
            style={
              Array [
                Object {
                  "fontSize": 27,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#484848",
                },
                Object {
                  "fontWeight": "500",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                null,
              ]
            }
            testID={undefined}
          >
            *
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#484848",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={undefined}
            style={
              Array [
                Object {
                  "fontSize": 27,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#484848",
                },
                Object {
                  "fontWeight": "500",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                null,
              ]
            }
            testID={undefined}
          >
            *
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#484848",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={undefined}
            style={
              Array [
                Object {
                  "fontSize": 27,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#484848",
                },
                Object {
                  "fontWeight": "500",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                null,
              ]
            }
            testID={undefined}
          >
            *
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#484848",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={undefined}
            style={
              Array [
                Object {
                  "fontSize": 27,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#484848",
                },
                Object {
                  "fontWeight": "500",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                null,
              ]
            }
            testID={undefined}
          >
            *
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#484848",
              "borderBottomWidth": 4,
              "marginRight": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={undefined}
            style={
              Array [
                Object {
                  "fontSize": 27,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#484848",
                },
                Object {
                  "fontWeight": "500",
                },
                null,
                Object {
                  "textAlign": "center",
                },
                null,
              ]
            }
            testID={undefined}
          >
            *
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "backgroundColor": "transparent",
              "color": "#484848",
            },
            Object {
              "fontWeight": "500",
            },
            null,
            Object {
              "textAlign": "center",
            },
            null,
          ]
        }
        testID={undefined}
      >
        Use PIN Code for "connect.me"
      </Text>
    </View>
  </View>
</View>
`;
